torch/_inductor/codegen/cpp.py

```diff
@@ -1397,7 +1397,7 @@ def masked(mask, body, other):
         assert isinstance(V.kernel, CppVecKernel)
         code = BracesBuffer()
         var = V.kernel.cse.newvar()
-        with V.kernel.masked(mask) as new_mask:
+        with V.kernel.masked(mask) as new_mask, V.kernel.bool_vectors_are_masks(False):
```

Comment:
I don't think this will work if we have kernels that mix bools and floats or other vectorized dtypes. Can you give a minimal reproducer for the failure with the current codegen?

Reply:
```python
import torch
cfn = torch.compile(torch.constant_pad_nd)
cfn(torch.zeros(5, dtype=torch.bool), [2, 3])
```

Reply:
You are right. This is not the correct solution. I'll open an issue